sda-commons-server-opentelemetry/src/main/java/org/sdase/commons/server/opentelemetry/OpenTelemetryBundle.java

```diff
@@ -0,0 +1,187 @@
+    return AutoConfiguredOpenTelemetrySdk.builder()
+        .addPropertiesSupplier(SdaConfigPropertyProvider::getProperties)
+        .setResultAsGlobal(shouldSetAsGlobal())
+        .build()
+        .getOpenTelemetrySdk();
+  }
+
+  public static InitialBuilder builder() {
+    return new Builder();
+  }
+
+  public interface InitialBuilder {
+    /**
+     * Specifies a custom telemetry instance to use. If this builder is used, this module will only
+     * be a consumer, and the provider needs to take care of registering this instance as global.
+     *
+     * @param openTelemetry The telemetry instance to use.
+     * @return the same builder
+     */
+    FinalBuilder withTelemetryInstance(OpenTelemetry openTelemetry);
+
+    /**
+     * Enables the bundle to setup an auto configured instance of OpenTelemetry Sdk, and register it
+     * as Global.
+     *
+     * @return the same builder.
+     */
+    FinalBuilder withAutoConfiguredTelemetryInstance();
+  }
+
+  public interface FinalBuilder {
+    /**
+     * Specifies a pattern where all traces for the matching urls will be suppressed. By default all
+     * requests with any url is traced.
+     *
+     * @param pattern The url pattern to be excluded.
+     * @return the same builder
+     */
+    FinalBuilder withExcludedUrlsPattern(Pattern pattern);
+
+    OpenTelemetryBundle build();
+  }
+
+  public static class Builder implements FinalBuilder, InitialBuilder {
+
+    private Supplier<OpenTelemetry> openTelemetryProvider;
+    private Pattern excludedUrlPatterns;
+
+    private Builder() {}
+
+    @Override
+    public FinalBuilder withTelemetryInstance(OpenTelemetry openTelemetry) {
+      this.openTelemetryProvider = () -> openTelemetry;
+      return this;
+    }
+
+    @Override
+    public FinalBuilder withExcludedUrlsPattern(Pattern excludedUrlPatterns) {
+      this.excludedUrlPatterns = excludedUrlPatterns;
+      return this;
+    }
+
+    @Override
+    public FinalBuilder withAutoConfiguredTelemetryInstance() {
+      if (shouldSkipTracing()) {
+        this.openTelemetryProvider = OpenTelemetry::noop;
+        return this;
+      }
+      this.openTelemetryProvider = OpenTelemetryBundle::bootstrapConfiguredTelemetrySdk;
+      return this;
+    }
+
+    @Override
+    public OpenTelemetryBundle build() {
+      return new OpenTelemetryBundle(openTelemetryProvider.get(), excludedUrlPatterns);
+    }
+
+    private static boolean shouldSkipTracing() {
+      // Skip loading the agent if tracing is disabled.
+      String jaegerSamplerType = getProperty(JAEGER_SAMPLER_TYPE);
+      String jaegerSamplerParam = getProperty(JAEGER_SAMPLER_PARAM);
+      boolean isDisabledUsingLegacyVars =
+          "const".equals(jaegerSamplerType) && "0".equals(jaegerSamplerParam);
+      if (isDisabledUsingLegacyVars) {
+        LOG.warn("Tracing is disabled using deprecated configuration.");
+        return true;
+      }
+      return false;
```

Comment:
You've only checked the legacy case. Do we need the current approach here as well using the `OTEL_` env variable?

Reply:
Added here, it was internally handled by `OTEL_EXPERIMENTAL_SDK_ENABLED ` changed to `OTEL_DISABLED`